### PR TITLE
Bug 1878648: oauth-apiserver: audit log oauthaccesstokens if fresh 4.6 install

### PR DESCRIPTION
**Pull request by sttts**
Counterpart to https://github.com/openshift/cluster-authentication-operator/pull/324.

The annotation `oauth-apiserver.openshift.io/secure-token-storage: true` set on the config/v1 apiserver object make openshift-apiserver deeply log oauth tokens, noth access and authorize (if openshift-apiserver is responsible for this; in fresh 4.6 install it is not, but oauth-apiserver is). 

This requires that there are no non-sha256 tokens in the system. For 4.6 installs this is guaranteed. 

After downgrade to 4.5 there is a controller removing the annotation. 

After upgrade from 4.5, the annotation is not set. The admin can set it manually after having made sure that no non-sha256 tokens exist.

For upgrade to 4.7, non-sha256 tokens must not exist. If they do, the upgrade will be blocked by the auth operator (to be implemented).

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -280,14 +280,27 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		return err
 	}
 
+	apiServerConfig, err := configClient.ConfigV1().APIServers().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(4.7): switch over to secure access-token logging by default and delete old non-sha256 tokens
```
Comment (tkashem) [inline pkg/operator/starter.go:288]: @sttts maybe we should create a BZ for this and set target release to 4.7 so we can track it?

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -280,14 +280,27 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
...
+	auditPolicyPathGetterWithAccessTokenLogs := func(profile string) (string, error) {
+		if apiServerConfig.Annotations["oauth-apiserver.openshift.io/secure-oauth-token-storage"] == "true" {
```
Comment (tkashem) [inline pkg/operator/starter.go:290]: what if the user removes the annotation from the APIServer object or set it to false? The `apiServerConfig` object is stale once it is changed by the user. May be I am misunderstanding the flow here?

File: pkg/operator/starter.go (hunk 1 above)
Comment (tkashem) [reply pkg/operator/starter.go:288]: @sttts I created this BZ to track this: https://bugzilla.redhat.com/show_bug.cgi?id=1879182

Comment (sttts) [reply pkg/operator/starter.go:288]: :+1:

File: pkg/operator/starter.go (hunk 2 above)
Comment (sttts) [reply pkg/operator/starter.go:290]: it will now redeploy with the old policy.

**Pull request merged**